added stability text-to-image job

gjohnsx opened the pull request:
I added a text-to-image job using the [Stability AI REST API](https://platform.stability.ai/docs/api-reference#tag/v1generation/operation/textToImage).

I wasn't sure what to do with the image in the example job so I'm just logging the huge base64 url

<img width="960" alt="image" src="https://github.com/triggerdotdev/api-reference/assets/93215236/b1160424-904d-451d-a677-ab18fb84a947">

If you have any other ideas for what to do with the artifacts once returned I'm all ears!

nicktrn commented on src/stability.ts:
```diff
@@ -0,0 +1,100 @@
+    const images = await io.runTask(
+      "Create image from text",
+      async () => {
+        const response = await fetch(
```
I think we should use `io.backgroundFetch()` for the reasons explained [here](https://trigger.dev/docs/sdk/io/backgroundfetch). This can be used outside of `io.runTask()` so we don't create an unnecessary subtask.

nicktrn commented on src/stability.ts:
```diff
@@ -0,0 +1,100 @@
+      images.artifacts.forEach((image, index) => {
+        const imageUrl = `data:image/png;base64,${image.base64}`;
+        // Log the URL to the console
+        io.logger.info(`Image ${index + 1}/${payload.samples ?? 1}:`, {
```
This is a great example of how to correctly use cacheKeys within loops! Will be useful for people to adapt to their needs.

The only issue here is the use of `forEach()`. This will not work as expected, especially with real subtasks. We should probably use async iteration with `for (const artifact of images.artifacts) { await ... }` or alternatively `await Promise.all(images.artifacts.map(..))`.

nicktrn reviewed (changes requested):
Thanks @gjohnsx, thanks for your contribution!

Think you forgot to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword). If you edit your post to add `closes #47` or `fixes #47` the issue can be automatically closed on merge. Makes things easier on our end! :pray: 

Left some comments.

gjohnsx commented:
@nicktrn thank you very much for the tips and for the comments! 

I changed the first io.runTask to io.backgroundFetch. 
Then I changed the way we iterate through the returned artifacts and we use a custom cache key inside the loop. We are just returning a URL of the image in this example. 

Take a look and let me know if that looks good.
<img width="960" alt="image" src="https://github.com/triggerdotdev/api-reference/assets/93215236/59190d9b-0d22-4df9-8d7f-8bf7d91822c5">

Test payload:
```
{
  "width": 1024,
  "height": 1024,
  "samples": 2,
  "text_prompts": [
    {
      "text": "A lighthouse on a cliff"
    }
  ]
}
```

nicktrn commented:
> I changed the first io.runTask to io.backgroundFetch

Looks good! You can make things even simpler by removing the typecast and passing it like this:

```ts
io.backgroundFetch<GenerationResponse>(...)
```

> We are just returning a URL of the image in this example

Sure, that's fine. My main concern was people adapting the previous example without realising they're creating (async) tasks within a `forEach()` loop. Appreciate the comment linking to the cache key docs.
